android/java/org/chromium/chrome/browser/playlist/PlaylistHostActivity.java

```diff
@@ -54,13 +57,16 @@
 public class PlaylistHostActivity extends AsyncInitializationActivity
         implements ConnectionErrorHandler, PlaylistOptionsListener,
                    PlaylistServiceObserverImplDelegate {
-    private static final String TAG = "Playlist/PlaylistHostActivity";
+    private static final String TAG = "PlaylistHostActivity";
     private PlaylistService mPlaylistService;
     private PlaylistViewModel mPlaylistViewModel;
     private PlaylistServiceObserverImpl mPlaylistServiceObserver;
 
     @Override
     public void onConnectionError(MojoException e) {
+        if (mPlaylistService != null) {
+            mPlaylistService.close();
```

Comment:
Shouldn't we also nullify the variable after closing it? This would help us in case `onConnectionError` is called twice.

```java
if (mPlaylistService != null) {
    mPlaylistService.close();
    mPlaylistService = null;
...
```